### PR TITLE
Optional join bug

### PR DESCRIPTION
**Pull request by darkfrog26**
Fixes a bug in querying with optional joins returning null.

### CODE REVIEW

File: core/src/main/scala/org/scalarelational/column/ColumnOption.scala [hunk 1]
```diff
@@ -11,6 +12,6 @@ case class ColumnOption[T, S](column: ColumnLike[T, S]) extends ColumnLike[Optio
   def dataType: DataType[Option[T], S] = OptionDataTypeCreator.create[T, S](column.dataType)
   def manifest: Manifest[Option[T]] = column.manifest.asInstanceOf[Manifest[Option[T]]]
```
Comment (tindzk) [inline core/src/main/scala/org/scalarelational/column/ColumnOption.scala]: Do we still need `manifest`?

File: core/src/main/scala/org/scalarelational/result/QueryResult.scala [hunk 2]
```diff
@@ -7,19 +7,18 @@ import org.scalarelational.table.Table
...
-  def apply[T, S](column: Column[T, S]): T = {
+  def apply[T, S](column: ColumnLike[T, S]): T = {
     get[T, S](column).getOrElse(throw new RuntimeException(s"Unable to find column: ${column.name} in result."))
   }
 
-  def has[T, S](column: Column[T, S]): Boolean = {
+  def has[T, S](column: ColumnLike[T, S]): Boolean = {
     val value = get[T, S](column)
-    value.nonEmpty && value.get != None
+    value.nonEmpty && value.get != None && !value.contains(null)
```
Comment (tindzk) [inline core/src/main/scala/org/scalarelational/result/QueryResult.scala]: `value.nonEmpty && value.get != None` can be shortened to just `value.nonEmpty`.

File: core/src/main/scala/org/scalarelational/result/QueryResult.scala [hunk 3]
```diff
@@ -7,19 +7,18 @@ import org.scalarelational.table.Table
 
 import scala.language.existentials
 
-
 case class QueryResult(table: Table, values: Vector[ExpressionValue[_]]) {
-  def get[T, S](column: Column[T, S]): Option[T] = values.collectFirst {
-    case cv: ColumnValue[_, _] if cv.column == column => Option(cv.value.asInstanceOf[T])
```
Comment (tindzk) [inline core/src/main/scala/org/scalarelational/result/QueryResult.scala]: Why did you change this? `Option(null)` becomes `None`.

File: core/src/main/scala/org/scalarelational/column/ColumnOption.scala (hunk 1 above)
Comment (darkfrog26) [inline core/src/main/scala/org/scalarelational/column/ColumnOption.scala]: Unfortunately we currently do for detecting "Option".  I'm not sure if there's any other use anymore.

File: core/src/main/scala/org/scalarelational/result/QueryResult.scala (hunk 3 above)
Comment (darkfrog26) [inline core/src/main/scala/org/scalarelational/result/QueryResult.scala]: Try changing it back and see what happens.
